src/CachePlugin.php

```diff
@@ -246,16 +244,23 @@ private function calculateResponseExpiresAt($maxAge)
     /**
      * Verify that we can cache this response.
      *
+     * @param RequestInterface  $request
      * @param ResponseInterface $response
      *
      * @return bool
      */
-    protected function isCacheable(ResponseInterface $response)
+    protected function isCacheable(RequestInterface $request, ResponseInterface $response)
```

Comment:
i wonder why this method is protected. extending the plugin sounds like a bad idea, but maybe that was the intention. as it is protected, this is technically a BC break. we could separate this into another method `isCacheableRequest` to avoid the BC break.

Reply:
you are right. i outsourced the checking of the request into the method isCacheableRequest and reset the "isCacheable" to it's original state.

The only thing, I am bothering about is the naming (one of the two hard things of computer science). If we add a method "isCacheableRequest", the method "isCacheable" should be renamed to "isCacheableResponse" and a new method "isCacheable" should take both arguments and return the result of isCacheableRequest && isCachableResponse().

This however would lead again to the BC, we would like to avoid.

Reply:
yep. my idea would be that in version 2 we make the plugin final, this method private and can change the signature.